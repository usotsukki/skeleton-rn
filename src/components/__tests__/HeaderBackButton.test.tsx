import { render } from '@testing-library/react-native'
import React from 'react'
import { ButtonDriver } from 'react-native-ui-lib/testkit'
import BackButton from '../HeaderBackButton'

const mockedNavigation = jest.fn()
const mockedGetState = jest.fn().mockReturnValue({ index: 0 })

jest.mock('expo-router', () => {
	return {
		useNavigation: () => ({
			goBack: mockedNavigation,
			getState: mockedGetState,
		}),
	}
})

describe('BackButton', () => {
	afterEach(() => {
		jest.clearAllMocks()
	})

	it(`doesn't render for first screen in stack`, () => {
		const renderTree = render(<BackButton />)
		const buttonNode = ButtonDriver({ renderTree, testID: 'nav-back-button' })

		expect(buttonNode.exists()).toBe(false)

		expect(renderTree.toJSON()).toBeNull()
	})

	it(`renders correctly for second screen in stack`, () => {
		mockedGetState.mockReturnValueOnce({ index: 1 })
		const renderTree = render(<BackButton />)
		const buttonNode = ButtonDriver({ renderTree, testID: 'nav-back-button' })

		expect(buttonNode.exists()).toBe(true)
		buttonNode.press()
		expect(mockedNavigation).toHaveBeenCalled()

		expect(renderTree.toJSON()).toMatchSnapshot()
	})
})
