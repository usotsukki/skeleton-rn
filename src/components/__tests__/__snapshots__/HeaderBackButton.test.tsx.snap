// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders correctly for second screen in stack 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 999,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": undefined,
      "opacity": 1,
      "paddingHorizontal": undefined,
      "paddingVertical": undefined,
    }
  }
  testID="nav-back-button"
>
  <Ionicons
    color="black"
    name="arrow-back"
    size={24}
  />
</View>
`;
